### ISSUE TITLE
Setup registry mirror integration test

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#2197]
We need to test the registry mirror option as part of the integration test. For this we need:

  * Docker on the CI node
  * pull the registry image
  * push configuration to container
  * start Minishift with registry mirror
  * check catalog of the registry for the image

A similar howto for setup on Windows: https://gist.github.com/gbraad/7cb36210bd5effdeb788c34d6b433ac8

**Comment by LalatenduMohanty** [comment]
@gbraad What is the intent behind the test? Is it only to make sure minishift works well with a registry mirror?

**Comment by gbraad** [comment]
yep, as it can aid in a disconnected scenario, but also a common usecase in slower and/or restricted networks; like corporate or China
